src/overlays/gamestates/ovl_file_choose/file_select.h

```diff
@@ -0,0 +1,335 @@
+    /*  2 */ FS_WARNING_NO_EMPTY_FILES, // "There is no empty file."
+    /*  3 */ FS_WARNING_FILE_EMPTY,     // "This is an empty file."
+    /*  4 */ FS_WARNING_FILE_IN_USE     // "This file is in use."
+} WarningLabel;
+
+typedef enum {
+    /* 0 */ FS_BTN_MAIN_FILE_1,
+    /* 1 */ FS_BTN_MAIN_FILE_2,
+    /* 2 */ FS_BTN_MAIN_FILE_3,
+    /* 3 */ FS_BTN_MAIN_COPY,
+    /* 4 */ FS_BTN_MAIN_ERASE,
+    /* 5 */ FS_BTN_MAIN_OPTIONS
+} MainMenuButtonIndex;
+
+typedef enum {
+    /* 0 */ FS_BTN_COPY_FILE_1,
+    /* 1 */ FS_BTN_COPY_FILE_2,
+    /* 2 */ FS_BTN_COPY_FILE_3,
+    /* 3 */ FS_BTN_COPY_QUIT
+} CopyMenuButtonIndex;
+
+typedef enum {
+    /* 0 */ FS_BTN_ERASE_FILE_1,
+    /* 1 */ FS_BTN_ERASE_FILE_2,
+    /* 2 */ FS_BTN_ERASE_FILE_3,
+    /* 3 */ FS_BTN_ERASE_QUIT
+} EraseMenuButtonIndex;
+
+typedef enum {
+    /* 0 */ FS_BTN_SELECT_FILE_1,
+    /* 1 */ FS_BTN_SELECT_FILE_2,
+    /* 2 */ FS_BTN_SELECT_FILE_3,
+    /* 3 */ FS_BTN_SELECT_YES,
+    /* 4 */ FS_BTN_SELECT_QUIT
+} SelectMenuButtonIndex;
+
+typedef enum {
+    /* 0 */ FS_BTN_CONFIRM_YES,
+    /* 1 */ FS_BTN_CONFIRM_QUIT
+} ConfirmButtonIndex;
+
+typedef enum {
+    /* 0 */ FS_BTN_ACTION_COPY,
+    /* 1 */ FS_BTN_ACTION_ERASE
+} ActionButtonIndex;
+
+typedef enum {
+    /* 0 */ FS_SETTING_AUDIO,
+    /* 1 */ FS_SETTING_ZTARGET
+} SettingIndex;
+
+typedef enum {
+    /* 0 */ FS_CHAR_PAGE_HIRA,
+    /* 1 */ FS_CHAR_PAGE_KATA,
+    /* 2 */ FS_CHAR_PAGE_ENG
+} CharPage;
+
+typedef enum {
+    /*  0 */ FS_KBD_BTN_HIRA,
+    /*  1 */ FS_KBD_BTN_KATA,
+    /*  2 */ FS_KBD_BTN_ENG,
+    /*  3 */ FS_KBD_BTN_BACKSPACE,
+    /*  4 */ FS_KBD_BTN_END,
+    /* 99 */ FS_KBD_BTN_NONE = 99
+} KeyboardButton;
+
+typedef struct FileSelectState {
+    /* 0x00000 */ GameState state;
+    /* 0x000A4 */ Vtx* windowVtx;
+    /* 0x000A8 */ u8* staticSegment;
+    /* 0x000AC */ u8* parameterSegment;
+    /* 0x000B0 */ u8* titleSegment;
+    /* 0x000B8 */ View view;
+    /* 0x00220 */ SramContext sramCtx;
+    /* 0x00248 */ SkyboxContext skyboxCtx;
+    /* 0x00470 */ MessageContext msgCtx;
+    /* 0x12550 */ Font font;
+    /* 0x242E0 */ EnvironmentContext envCtx;
+    /* 0x243E0 */ Vtx* unk_243E0;
+    /* 0x243E4 */ Vtx* windowContentVtx;
+    /* 0x243E8 */ Vtx* keyboardVtx;
+    /* 0x243EC */ Vtx* nameEntryVtx;
+    /* 0x243F0 */ Vtx* keyboard2Vtx;
+    /* 0x243F4 */ u8 newf[4][6];
+    /* 0x2440C */ u16 sotCount[2];
+    /* 0x24410 */ u16 sotCount2[2];
+    /* 0x24414 */ char fileNames[2][8];
+    /* 0x24424 */ char fileNames2[2][8];
+    /* 0x24434 */ s16 healthCapacity[2];
+    /* 0x24438 */ u16 healthCapacity2[2];
+    /* 0x2443C */ s16 health[2];
+    /* 0x24440 */ u16 health2[2];
+    /* 0x24444 */ u32 questItems[2];
+    /* 0x2444C */ u32 questItems2[2];
+    /* 0x24454 */ s8 defenseHearts[2];
+    /* 0x24456 */ u8 defenseHearts2[2];
+    /* 0x24458 */ u16 time[2];
+    /* 0x2445C */ u16 time2[2];
+    /* 0x24460 */ s16 day[2];
+    /* 0x24464 */ s16 day2[2];
+    /* 0x24468 */ u8 isOwlSave[2];
```

Comment:
Is this used elsewhere? Only the second seems to be used in file_choose.

Reply:
Same as previous comment, why there's `isOwlSave` vs. `isOwlSave2` isn't clear and needs more research/decomp? who knows if it shows up elsewhere